### PR TITLE
feat(sdk): easily generate container urns

### PR DESCRIPTION
**Pull request by hsheth2**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/emitter/mce_builder.py [hunk 1]
```diff
@@ -132,7 +145,11 @@ def dataset_key_to_urn(key: DatasetKeyClass) -> str:
     )
 
 
-def make_container_urn(guid: str) -> str:
+def make_container_urn(guid: Union[str, "DatahubKey"]) -> str:
```
Comment (shirshanka) [inline metadata-ingestion/src/datahub/emitter/mce_builder.py:148]: Is the idea that you must be using one of these concrete classes like DatabaseKey etc. that extend DataHubKey?
So each container entity type must create a concrete Key class?

Comment (hsheth2) [reply metadata-ingestion/src/datahub/emitter/mce_builder.py:148]: effectively yes - we already have concrete key classes across our sources, and I much prefer the real classes over passing a property bag dict around everywhere

File: metadata-ingestion/src/datahub/emitter/mcp_builder.py [hunk 2]
```diff
@@ -81,6 +81,9 @@ def guid_dict(self) -> Dict[str, str]:
     def property_dict(self) -> Dict[str, str]:
         return self.dict(by_alias=True, exclude_none=True)
 
+    def as_urn(self) -> str:
+        return make_container_urn(guid=self.guid())
```
Comment (asikowitz) [inline metadata-ingestion/src/datahub/emitter/mcp_builder.py:85]: So are all these `DatahubKey`s really container keys? Can we call them that? `PlatformKey` makes me think we're creating a key for a platform... but really we're creating a key for a container that contains platform information.

**Review by asikowitz — APPROVED** [review]
One naming request but looks reasonable enough to me

File: metadata-ingestion/src/datahub/emitter/mcp_builder.py (hunk 2 above)
Comment (hsheth2) [reply metadata-ingestion/src/datahub/emitter/mcp_builder.py:87]: agree that it's confusing - going to rename PlatformKey to ContainerKey + add some docs

**Pull request merged**